docs/explanation/e-hardening.md

```diff
@@ -0,0 +1,119 @@
+
+### Juju 
+
+Juju is the component responsible for orchestrating the entire lifecycle, from deployment to Day 2 operations, of 
+all applications. Therefore, it is imperative that it is set up securely. Please refer to the Juju documentation for more information on:
+
+* [Juju security](https://discourse.charmhub.io/t/juju-security/15684)
+* [How to harder your deployment](https://juju.is/docs/juju/harden-your-deployment)
+
+#### Cloud credentials
+
+When configuring the cloud credentials to be used with Juju, ensure that the users have correct permissions to operate at the required level. 
+Juju superusers responsible for bootstrapping and managing controllers require elevated permissions to manage several kind of resources, such as
+virtual machines, networks, storages, etc. Please refer to the references below for more information on the policies required to be used depending on the cloud. 
+
+| Cloud     | Cloud user policies                                                                                                                                                                                                                            |
+|-----------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| OpenStack |                                                                                                                                                                                                                                                |
+| AWS       | [Juju AWS Permission](https://discourse.charmhub.io/t/juju-aws-permissions/5307), [AWS Instance Profiles](https://discourse.charmhub.io/t/using-aws-instance-profiles-with-juju-2-9/5185), [Juju on AWS](https://juju.is/docs/juju/amazon-ec2) | 
+| Azure     | [Juju Azure Permission](https://juju.is/docs/juju/microsoft-azure), [How to use Juju with Microsoft Azure](https://discourse.charmhub.io/t/how-to-use-juju-with-microsoft-azure/15219)                                                                                                  |
+
+#### Juju users
+
+It is very important that the different juju users are set up with minimal permission depending on the scope for their operations. 
+Please refer to the [User access levels](https://juju.is/docs/juju/user-permissions) documentation for more information on the access level and corresponding abilities 
+that the different users can be granted. 
+
+Juju user credentials must be stored securely and rotated regularly to prevent leakage and unauthorized access. 
```

Comment:
```suggestion
Juju user credentials must be stored securely and rotated regularly to prevent unauthorized access. 
```

Reply:
why? The point of rotating credentials is related to the fact that credentials may leak

Reply:
Correct me if I'm wrong but my understanding is the following.
1. "Leak" is an unofficial term. Secrets can be compromised. 
2. Anyone else using your "leaked" credentials is performing unauthorized access.
3. You can't prevent a "leak" by rotating. You can prevent unauthorized access with the leaked credentials (by rotating post-factum either specifically to address compromised credentials or by schedule).

Reply:
Yes, that's fine. I would then suggest:

> Juju user credentials must be stored securely and rotated regularly to limit chances of unauthorized access due to credentials leakage. 

Unauthorized access can happen for different reasons. I don't think we should claim that rotation prevent all unauthorized access.

Reply:
Looks much better this way. If you really want to put the word "leakage" there it is probably the best way to do it.
I'll add it as a proper suggestion because that enables me to approve this PR with the suggestion. And if this suggestion is merged it should not invalidate my approval.
```suggestion
Juju user credentials must be stored securely and rotated regularly to limit the chances of unauthorized access due to credentials leakage.
```
Alternatively, I'd consider using "due to compromised credentials".